app/components/primer/alpha/text_field.rb

```diff
@@ -0,0 +1,116 @@
+
+      # @!method initialize
+      #
+      # @example Default
+      #   <%= render(Primer::Alpha::TextField.new(name: :first_name, label: "First name")) %>
+      #
+      # @example With a trailing label
+      #   <%= render(
+      #     Primer::Alpha::TextField.new(
+      #       name: :days,
+      #       label: "Number of days",
+      #       show_label: false,
+      #       trailing_label: "days"
+      #     )
+      #   ) %>
+      #
+      # @example With a clear button
+      #   <%= render(
+      #     Primer::Alpha::TextField.new(
+      #       name: :first_name,
+      #       label: "First name",
+      #       show_clear_button: true
+      #     )
+      #   ) %>
+      #
+      # @example Full width
+      #   <%= render(
+      #     Primer::Alpha::TextField.new(
+      #       name: :first_name,
+      #       label: "First name",
+      #       full_width: true
+      #     )
+      #   ) %>
+      #
+      # @example Disabled
+      #   <%= render(
+      #     Primer::Alpha::TextField.new(
+      #       name: :first_name,
+      #       label: "First name",
+      #       disabled: true
+      #     )
+      #   ) %>
+      #
+      # @example Invalid
+      #   <%= render(
+      #     Primer::Alpha::TextField.new(
+      #       name: :first_name,
+      #       label: "First name",
+      #       invalid: true
+      #     )
+      #   ) %>
+      #
+      # @example With a leading visual
+      #   <%= render(
+      #     Primer::Alpha::TextField.new(
+      #       name: :first_name,
+      #       label: "First name",
+      #       leading_visual: {
+      #         icon: :person
+      #       }
+      #     )
+      #   ) %>
+      #
+      # @example With a caption
+      #   <%= render(
+      #     Primer::Alpha::TextField.new(
+      #       name: :first_name,
+      #       label: "First name",
+      #       caption: "What your friends call you"
+      #     )
+      #   ) %>
+      #
+      # @example With a validation message
+      #   <%= render(
+      #     Primer::Alpha::TextField.new(
+      #       name: :first_name,
+      #       label: "First name",
+      #       validation_message: "Hmm, that doesn't look right"
+      #     )
+      #   ) %>
+      #
+      # @param name [String] Value for the HTML name attribute.
+      # @param id [String] Value for the HTML id attribute.
+      # @param class [String] A list of CSS classes to add to the input. Exists for compatibility with Rails form builders.
+      # @param classes [String] A list of CSS classes to add to the input. Combined with the `:class` argument.
+      # @param caption [String] Caption text to render below the input.
+      # @param label [String] Label text displayed above the input.
+      # @param show_label [Boolean] Whether or not to visually display the label. If `true` the label will still be available to screen readers.
+      # @param trailing_label [String] Text to display to the right of the input.
```

Comment:
Do we have a use case to support this param? I don't think we want to support this design, so I would remove this option.

Reply:
I added this to support the "Artifact and Log Retention" form:

![image](https://user-images.githubusercontent.com/575280/182242746-fe1faaa7-a4bb-440d-b0dd-0f18a2b4b4ae.png)

However, I never ended up landing the migration for it, so we could remove it I guess?

Reply:
Ahh right. Lets remove it for now and re-evaluate. A similar case came up recently for Primer Brand, so we can align with that solution if they decide to add in a property.